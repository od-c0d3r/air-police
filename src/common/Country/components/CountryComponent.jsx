import { Card, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CountryComponent = ({ country }) => {
  const { data: { main: { aqi } } } = country;
  const idxs = {
    1: 'Good',
    2: 'Fair',
    3: 'Moderate',
    4: 'Poor',
    5: 'Very Poor',
  };
  const classNames = {
    1: 'one',
    2: 'two',
    3: 'three',
    4: 'four',
    5: 'five',
  };

  const rate = idxs[Number(aqi)];
  const getClassName = classNames[Number(aqi)];

  return (
    <Col>
      <Link variant="primary" className="cardLink" to={`/country/${country.id}`}>
        <Card className="shadow-sm">
          <Card.Body>
            <Card.Title>{country.name}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">
              {`AQI: ${aqi}`}
            </Card.Subtitle>
            <Card.Text>
              <span className="d-block">
                Overall Quailty :
              </span>
              <span id="rate">
                <span className={getClassName}>
                  {rate}
                </span>
              </span>
            </Card.Text>
          </Card.Body>
        </Card>
      </Link>
    </Col>
  );
};

CountryComponent.propTypes = {
  country: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    data: PropTypes.shape([]),
  }).isRequired,
};

export default CountryComponent;
