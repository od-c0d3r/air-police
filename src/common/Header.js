import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <Link to="/">HOME</Link>
    {' '}
    |
    {' '}
  </header>
);

export default Header;
