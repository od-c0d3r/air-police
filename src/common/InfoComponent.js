import { useState } from 'react';
import { Alert, Button } from 'react-bootstrap';
import { HiOutlineExclamationCircle } from 'react-icons/hi';

const InfoComponent = () => {
  const [show, setShow] = useState(true);

  return (
    <>
      <Alert id="alert-window" show={show} variant="success">
        <Alert.Heading>How&apos;s it going?!</Alert.Heading>
        <p>
          Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget
          lacinia odio sem nec elit. Cras mattis consectetur purus sit amet
          fermentum.
        </p>
        <hr />
        <div className="d-flex justify-content-end">
          <Button onClick={() => setShow(false)}>Close Alert</Button>
        </div>
      </Alert>

      {!show && <HiOutlineExclamationCircle id="info" onClick={() => setShow(true)} />}
    </>
  );
};

export default InfoComponent;
