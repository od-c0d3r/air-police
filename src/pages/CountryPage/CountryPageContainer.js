const CountryPageContainer = () => <h1>#CountryPageContainer</h1>;

export default CountryPageContainer;
