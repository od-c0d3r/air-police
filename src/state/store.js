import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger'
import { reduxBatch } from '@manaflair/redux-batch'
import airReducer from './air/airReducer';

export const store = configureStore({
  reducer: {
    air: airReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
  enhancers: [reduxBatch],
});
