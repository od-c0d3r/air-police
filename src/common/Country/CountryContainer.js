import Row from 'react-bootstrap/Row';
import { useSelector } from 'react-redux';
import CountryComponent from './components/CountryComponent';

const CountryContainer = () => {
  const data = useSelector((state) => state.airApp.filter);

  return (
    <Row xs={2} md={3} className="g-4 ms-0 me-0">
      {data.map((country) => (
        <CountryComponent
          key={country.id}
          country={country}
        />
      ))}
    </Row>
  );
};

export default CountryContainer;
