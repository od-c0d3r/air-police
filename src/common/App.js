import React from 'react';
import Header from './Header';
import createRoutes from '../routes';

function App() {
  return (
    <main id="App">
      <Header />
      {createRoutes()}
    </main>
  );
}

export default App;
