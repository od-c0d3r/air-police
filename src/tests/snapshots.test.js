import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../state/store';
import App from '../common/App'
import Header from '../common/Header';
import CountryContainer from '../common/Country/CountryContainer';
import CountryComponent from '../common/Country/components/CountryComponent';
import InfoComponent from '../common/InfoComponent';
import CountryPageContainer from '../pages/CountryPage/CountryPageContainer';

jest.mock('../utils/filter');

describe('Snapshots', () => {
    xit('<App />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <App />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('<Header />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('<InfoComponent />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <InfoComponent />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('<CountryContainer />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <CountryContainer />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    xit('<CountryComponent />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <CountryComponent />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('<CountryPageContainer />', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BrowserRouter>
                        <CountryPageContainer />
                    </BrowserRouter>
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
})
