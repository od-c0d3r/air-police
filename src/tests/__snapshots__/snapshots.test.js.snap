// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots <App /> 1`] = `
<main
  id="App"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand"
      >
        <a
          aria-current="page"
          className="navLink active"
          href="/"
          onClick={[Function]}
        >
          AirPolice
        </a>
      </span>
      <button
        aria-controls="navbarScroll"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <form
        className="d-flex"
        onSubmit={[Function]}
      >
        <input
          aria-label="🔍"
          className="me-2 form-control"
          id="inputSearch"
          onChange={[Function]}
          placeholder="Filter by Country Name"
          type="search"
        />
      </form>
    </div>
  </nav>
  <div
    className="g-4 ms-0 me-0 row row-cols-md-3 row-cols-2"
  />
  <svg
    data-testid="info"
    fill="none"
    height="1em"
    id="info"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
</main>
`;

exports[`Snapshots <CountryContainer /> 1`] = `
<div
  className="g-4 ms-0 me-0 row row-cols-md-3 row-cols-2"
/>
`;

exports[`Snapshots <CountryPageContainer /> 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="display-1 col"
    >
      Egypt
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      AQI
    </div>
    <div
      className="col"
    >
      5
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Co
    </div>
    <div
      className="col"
    >
      247
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      NH3
    </div>
    <div
      className="col"
    >
      0
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      No
    </div>
    <div
      className="col"
    >
      0.01
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      No2
    </div>
    <div
      className="col"
    >
      0.19
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      O3
    </div>
    <div
      className="col"
    >
      82.97
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Pm10
    </div>
    <div
      className="col"
    >
      668.37
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      So2
    </div>
    <div
      className="col"
    >
      1.03
    </div>
  </div>
</div>
`;

exports[`Snapshots <Header /> 1`] = `
<nav
  className="navbar navbar-expand navbar-light bg-light"
>
  <div
    className="container-fluid"
  >
    <span
      className="navbar-brand"
    >
      <a
        aria-current="page"
        className="navLink active"
        href="/"
        onClick={[Function]}
      >
        AirPolice
      </a>
    </span>
    <button
      aria-controls="navbarScroll"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <form
      className="d-flex"
      onSubmit={[Function]}
    >
      <input
        aria-label="🔍"
        className="me-2 form-control"
        id="inputSearch"
        onChange={[Function]}
        placeholder="Filter by Country Name"
        type="search"
      />
    </form>
  </div>
</nav>
`;

exports[`Snapshots <InfoComponent /> 1`] = `
<svg
  data-testid="info"
  fill="none"
  height="1em"
  id="info"
  onClick={[Function]}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 24 24"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;
